heron/schedulers/src/java/org/apache/heron/scheduler/kubernetes/V1Controller.java

```diff
@@ -1101,80 +1104,212 @@ protected V1ConfigMap getConfigMap(String configMapName) {
    */
   @VisibleForTesting
-  protected Pair<List<V1Volume>, List<V1VolumeMount>> createPersistentVolumeClaimVolumesAndMounts(
-      final Map<String, Map<KubernetesConstants.VolumeClaimTemplateConfigKeys, String>> mapConfig) {
-    List<V1Volume> volumeList = new LinkedList<>();
-    List<V1VolumeMount> mountList = new LinkedList<>();
-    for (Map.Entry<String, Map<KubernetesConstants.VolumeClaimTemplateConfigKeys, String>> configs
+  protected void createVolumeAndMountsPersistentVolumeClaimCLI(
+      Map<String, Map<KubernetesConstants.VolumeConfigKeys, String>> mapConfig,
+      List<V1Volume> volumes, List<V1VolumeMount> volumeMounts) {
+    for (Map.Entry<String, Map<KubernetesConstants.VolumeConfigKeys, String>> configs
         : mapConfig.entrySet()) {
       final String volumeName = configs.getKey();
-      final String path = configs.getValue()
-          .get(KubernetesConstants.VolumeClaimTemplateConfigKeys.path);
-      final String subPath = configs.getValue()
-          .get(KubernetesConstants.VolumeClaimTemplateConfigKeys.subPath);
-
-      if (path == null || path.isEmpty()) {
-        throw new TopologySubmissionException(
-            String.format("A mount path is required and missing from '%s'", volumeName));
-      }
 
       // Do not create Volumes for `OnDemand`.
       final String claimName = configs.getValue()
-          .get(KubernetesConstants.VolumeClaimTemplateConfigKeys.claimName);
+          .get(KubernetesConstants.VolumeConfigKeys.claimName);
       if (claimName != null && !KubernetesConstants.LABEL_ON_DEMAND.equalsIgnoreCase(claimName)) {
-        final V1Volume volume = new V1VolumeBuilder()
-            .withName(volumeName)
-            .withNewPersistentVolumeClaim()
-              .withClaimName(claimName)
-            .endPersistentVolumeClaim()
-            .build();
-        volumeList.add(volume);
+        volumes.add(
+            new V1VolumeBuilder()
+                .withName(volumeName)
+                .withNewPersistentVolumeClaim()
+                  .withClaimName(claimName)
+                .endPersistentVolumeClaim()
+                .build()
+        );
       }
+      volumeMounts.add(createVolumeMountsCLI(volumeName, configs.getValue()));
+    }
+  }
 
-      final V1VolumeMountBuilder volumeMount = new V1VolumeMountBuilder()
+  /**
+   * Generates the <code>Volume</code>s and <code>Volume Mounts</code> for <code>emptyDir</code>s to be
+   * placed in the <code>Executor</code> and <code>Manager</code> from options on the CLI.
+   * @param mapOfOpts Mapping of <code>Volume</code> option <code>key-value</code> configuration pairs.
+   * @param volumes A list of <code>Volume</code> to append to.
+   * @param volumeMounts A list of <code>Volume Mounts</code> to append to.
+   */
+  @VisibleForTesting
+  protected void createVolumeAndMountsEmptyDirCLI(
+      Map<String, Map<KubernetesConstants.VolumeConfigKeys, String>> mapOfOpts,
```

Comment:
Would it be worthwhile to mark this parameters passed in the function as `final`?

Reply:
Changes affected.